### ISSUE TITLE
Unable to deploy cluster to existing virtual network with existing cluster

### ISSUE THREAD

**Issue by JoeBerg8** [elastic/azure-marketplace#278]
All of the following is taking place in the same resource group:

The load balancers referred to below are all internal.

I have an existing virtual network which has a subnet with IPs in a designated address space.  I was able to successfully deploy a cluster there using the template, I'll refer to it as my **non-prod-cluster**.

I want to deploy a production cluster to the same virtual network, I'll refer to this as my **prod-cluster**.  This seems like it should work but I am running into issues in practice.  The **vNetLoadBalancerIP** parameter asks for an available IP on the virtual network, which I am supplying as the value.  When the template deploys, it actually replaces the **vNetLoadBalancerIP** from my **non-prod-cluster** with the value I provided when trying to deploy my **prod-cluster**, which then proceeds to fail the deployment entirely.

**TLDR**: I can't seem to figure out a way to deploy a new cluster to an existing virtual network while using a new, but also available **vNetLoadBalancerIP**.  Trying to do so overwrites the **vNetLoadBalancerIP** of my already deployed cluster.   This causes my deployment to fail.

Thanks in advance for any help!

**Comment by russcam** [comment]
@JoeBerg8 could you provide more detail about how you are deploying e.g. 

1. using Azure Marketplace solution template, or using the template directly from GitHub with Azure CLI/PowerShell?
2. What parameters are you supplying?

[Here's an example of deploying two clusters to one vnet](https://github.com/elastic/azure-marketplace-examples/blob/master/two_clusters_one_vnet/two_clusters_one_vnet.ps1). It uses a _very_ old version of the template now which I suspect would be broken because of the JDK dependencies no longer being available, but hopefully gives some idea of how to do this. The two crucial parts are

1. Providing an available IP address on the vnet for `vNetLoadBalancerIp`
2. Providing a unique hostname prefix for each cluster with `vmHostNamePrefix`

**Comment by JoeBerg8** [comment]
@russcam Hey Russ, sure thing.

I am using the template directly from GitHub with the Azure CLI.

My parameters are as follows:

`{
  "artifactsBaseUrl":{"value":"https://raw.githubusercontent.com/elastic/azure-marketplace/master/src"},
  "esVersion":{"value":"6.7.1"},
  "esClusterName":{"value":"prod-search-live"},
  "loadBalancerType":{"value":"internal"},
  "azureCloudPlugin":{"value":"No"},
  "azureCloudStorageAccountName":{"value":""},
  "azureCloudStorageAccountResourceGroup":{"value":""},
  "xpackPlugins":{"value":"Yes"},
  "esAdditionalPlugins": { "value":""},
  "esAdditionalYaml": { "value":""},
  "esHeapSize": { "value":0},
  "esHttpCertBlob": { "value":""},
  "esHttpCertPassword": { "value":""},
  "esHttpCaCertBlob": { "value":""},
  "esHttpCaCertPassword": { "value":""},
  "esTransportCaCertBlob": { "value":""},
  "esTransportCaCertPassword": { "value":""},
  "esTransportCertPassword": { "value":""},
  "samlMetadataUri": { "value":""},
  "samlServiceProviderUri": { "value":""},
  "kibana":{"value":"Yes"},
  "vmSizeKibana":{"value":"Standard_D2s_v3"},
  "vmKibanaAcceleratedNetworking":{"value":"Default"},
  "kibanaCertBlob":{"value":""},
  "kibanaKeyBlob":{"value":""},
  "kibanaKeyPassphrase":{"value":""},
  "kibanaAdditionalYaml": { "value":""},
  "logstash": { "value":"No"},
  "vmSizeLogstash": { "value":"Standard_D1"},
  "vmLogstashAcceleratedNetworking":{"value":"Default"},
  "logstashHeapSize": { "value": 0 },
  "logstashConf": { "value": "" },
  "logstashKeystorePassword": { "value": "" },
  "logstashAdditionalPlugins": { "value": "" },
  "logstashAdditionalYaml": { "value": "" },
  "jumpbox":{"value":"No"},
  "vmHostNamePrefix":{"value":"plive"},
  "vmSizeDataNodes":{"value":"Standard_F32s_v2"},
  "vmDataNodeAcceleratedNetworking":{"value":"Default"},
  "vmDataDiskCount":{"value":3},
  "vmDataDiskSize":{"value":"32GiB"},
  "vmDataNodeCount":{"value":10},
  "storageAccountType":{"value":"Default"},
  "dataNodesAreMasterEligible":{"value":"Yes"},
  "vmSizeMasterNodes":{"value":"Standard_DS2"},
  "vmMasterNodeAcceleratedNetworking":{"value":"Default"},
  "vmClientNodeCount":{"value":0},
  "vmSizeClientNodes":{"value":"Standard_D1"},
  "vmClientNodeAcceleratedNetworking":{"value":"Default"},
  "adminUsername":{"value":"admin"},
  "authenticationType":{"value":"password"},
  "sshPublicKey":{"value":"<value-ignored>"},
  "adminPassword":{"value":""},
  "securityBootstrapPassword":{"value":""},
  "securityAdminPassword":{"value":""},
  "securityKibanaPassword":{"value":""},
  "securityLogstashPassword":{"value":""},
  "securityBeatsPassword":{"value":""},
  "securityApmPassword":{"value":""},
  "securityRemoteMonitoringPassword":{"value":""},
  "vNetNewOrExisting": {"value":"existing"},
  "vNetName": {"value": "es-net"},
  "vNetExistingResourceGroup": {"value": "my-resource-group"},
  "vNetLoadBalancerIp": {"value": "10.233.125.10"},
  "vNetClusterSubnetName": {"value": "es-virtual-net"},
}
`

When attempting to deploy an additional cluster to an existing vnet the vnet load balancer IP address I provide overwrites the IP of the existing load balancer associated with an already deployed cluster.

I am also using a unique hostname prefix when deploying each time.

**Comment by russcam** [comment]
This provides the value `"10.233.125.10"` for the "prod-search-live" cluster internal load balancer IP. Some questions

1. Is this address available in the vnet?
2. What's the IP address of the existing cluster's internal load balancer IP?
3. Are you using Azure DNS?

I just updated [the 2 clusters 1 vnet example](https://github.com/elastic/azure-marketplace-examples/blob/master/two_clusters_one_vnet/two_clusters_one_vnet.ps1) and all worked as expected when deployed, so trying to see what might be different in your environment.

**Comment by JoeBerg8** [comment]
That IP is available in the vnet.  

The IP address of the existing cluster's internal load balancer is 10.233.125.25 - which is indeed available on the subnet.  The issue here is that when I give the load balancer a new and available IP for my second cluster, it overwrites the IP of the existing internal load balancer.  I should mention again that the deploy method I am using is with the azure CLI, and not the two clusters, one vnet script you have referenced. My deployment command looks something like this: `az group deployment create -g my-resource-group --template-uri https://raw.githubusercontent.com/elastic/azure-marketplace/master/src/mainTemplate.json --parameters @params_prod.json` where `params_prod.json` is the list of parameters posted in my above comment.

I am not using Azure DNS.

Thanks.

**Comment by russcam** [comment]
Is the existing vnet in its own resource group, or deployed with the first cluster?

I'm going to see if I can replicate it with Azure CLI

**Comment by JoeBerg8** [comment]
The existing vnet is not in its own resource group.  it is in the same resource group as the rest of the resources that make up the cluster. 

The script is in a sense, idempotent.  It will only update properties that change.  So if I leave all parameters the same but increase the number of data nodes, then the only resources that will be deployed are the new data nodes.  Likewise, changing the load balancer IP will cause the existing load balancer IP to be updated, and not deployed as a new load balancer.

**Comment by russcam** [comment]
Yes, the default deployment mode is incremental. 

Are you deploying the second cluster into the same resource group as the first cluster? If that is the case, I could see that being an issue because the internal load balancer name is not prefixed with the `vmHostNamePrefix` value, so to the Azure infrastructure, the second cluster deployment would look like it's making a change to the existing deployment's internal load balancer

**Comment by JoeBerg8** [comment]
I am deploying to the same resource group as the first cluster.  Looks like your PR will address the problem.  Thanks a ton for your help!
